### ISSUE TITLE
`reportUnknownArgumentType` false negative in `__init__` statements

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#746]
Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&reportUnknownParameterType=false&reportMissingParameterType=false&code=MYGwhgzhAECCB2BPA8gJwKrwNbwPYHd4AuAWAChpLoATAUwDNoB9JgS3lYBcWAKCWkPQA00AG5gQAV1oBKUhSqKADqnace4qbOgBiaHmi1UqXKnLlFdRvVy4%2BA4WInS5FxVRVqNz7XqMnUImhUWiVTTkwcAnhYVABzSQBbWnhOABVEJVogA)

```python
class AnyOrUnknown:
    def __init__(self, value):
        print(value) # no error

    def foo(self, value):
        print(value) # error: reportUnknownArgumentType
```

this seems to be because pyright creates a fake typevar for constructor arguments, no idea why

**Comment by DetachHead** [comment]
```py
class Bar:
    asdf: int = 1

class Foo:
    def __init__(self, value):
        self.value = value

reveal_type(Foo(Bar()).value) # Bar
```
